x-pack/filebeat/input/entityanalytics/internal/kvstore/input_test.go

```diff
@@ -0,0 +1,227 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package kvstore
+
+import (
+	"context"
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/paths"
+)
+
+var _ Input = &testInput{}
+
+type testInput struct {
+	name   string
+	testFn func(testCtx v2.TestContext) error
+	runFn  func(runCtx v2.Context, store *Store, client beat.Client) error
+}
+
+func (n *testInput) Name() string {
+	return n.name
+}
+
+func (n *testInput) Test(testCtx v2.TestContext) error {
+	if n.testFn != nil {
+		return n.testFn(testCtx)
+	}
+
+	return nil
+}
+
+func (n *testInput) Run(runCtx v2.Context, store *Store, client beat.Client) error {
+	if n.runFn != nil {
+		return n.runFn(runCtx, store, client)
+	}
+
+	return nil
+}
+
+var _ beat.Pipeline = &testPipeline{}
+
+type testPipeline struct {
+}
+
+func (t testPipeline) ConnectWith(_ beat.ClientConfig) (beat.Client, error) {
+	return &testClient{}, nil
+}
+
+func (t testPipeline) Connect() (beat.Client, error) {
+	return &testClient{}, nil
+}
+
+var _ beat.Client = &testClient{}
+
+type testClient struct {
+}
+
+func (c *testClient) Publish(_ beat.Event) {
+
+}
+
+func (c *testClient) PublishAll(_ []beat.Event) {
+
+}
+
+func (c *testClient) Close() error {
+	return nil
+}
+
+func TestInput_Name(t *testing.T) {
+	name := "testInput"
+	inp := input{
+		managedInput: &testInput{
+			name: name,
+		},
+	}
+
+	assert.Equal(t, name, inp.Name())
```

Comment:
> > Testing the input outside of unit tests requires access to Azure Active Directory.
> 
> I assume it was actually tested manually with AD, could you confirm that and describe the test steps and the example of the output?
> 
> Having just unit tests is not enough in this case.

Yes, this was tested manually with Azure AD. For a list of test cases, I ran through the following scenarios (this was linked in the related issue):

### Test Criteria

- [x] Verify users are synced.
- [x] Verify documents contain group membership info.
- [x] Update a user in Azure AD, verify the change is reflected in Elasticsearch as a new document.
- [x] Delete a user in Azure AD, verify the change is reflected in Elasticsearch as a new document that indicates a deleted status
- [x] Create a new user in Azure AD, verify the new user reflected in Elasticsearch
- [x] Verify that the data stream contains a "full sync marker" document with `event.action: started` when a new sync starts.
- [x] Verify that the data stream contains a "full sync marker" document with `event.action: completed` when a new sync completed.
- [x] Documentation exists that explains how the input works (what APIs it utilizes, how it persists information, etc)
- [x] Documentation exists that explains how to authenticate and authorize the input with least privileges.

It takes a while to get everything set up, and I just returned from traveling, so it may take me a bit to get sample documents.

Reply:
To be honest, first time I've heard of `require`. `assert` was always my go-to and many other tests in beats use it.

Is this a required change for all beats moving forward? If so, please make an announcement (and apologies if I missed it if one was made), so we can make changes ahead of time.